### PR TITLE
Statistics Infrastructure github clones

### PR DESCRIPTION
**Pull request by gurkanindibay**

### CODE REVIEW

File: packaging_automation/github_stats_collector.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+def fetch_and_store_github_clones(organization_name: str, repo_name: str, db_parameters: DbParams, github_token: str,
+                                  is_test: bool):
+    contents = github_clone_stats(github_token, organization_name, repo_name)
+
+    session = db_session(db_parameters, is_test)
+
+    fetch_time = datetime.now()
+
+    main_transaction = GithubCloneStatsTransactionsMain(fetch_time=fetch_time, count=contents['count'],
+                                                        repo_name=repo_name, uniques=contents['uniques'])
```
Comment (halilozanakgul) [inline packaging_automation/github_stats_collector.py:74]: I'm a bit confused here. Doesn't `get_clones_traffic` in `github_clone_stats` return a list of clone objects. Why would the contents include `count` and `unique`? Am I reading the docs wrong?

File: packaging_automation/github_stats_collector.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    for daily_record in contents['clones']:
+        # current date's record is skipped since statistics continue to change until end of the day
+        # stat record will not be added if it exists in
+        if daily_record.timestamp.date() == fetch_time.date() or stat_records_exists(daily_record.timestamp.date(),
+                                                                                     session=session):
+            continue
+        stats_record = GithubCloneStats(fetch_time=fetch_time, clone_date=daily_record.timestamp,
+                                        count=daily_record.count, uniques=daily_record.uniques,
+                                        repo_name=repo_name)
+        detail_transaction = GithubCloneStatsTransactionsDetail(clone_date=daily_record.timestamp,
```
Comment (halilozanakgul) [inline packaging_automation/github_stats_collector.py:84]: What is the purpose of the `GithubCloneStatsTransactionsDetail`? It feels like it is `GithubCloneStats` with less info.

File: packaging_automation/tests/test_github_stats_collector.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    session.delete(first_record)
+    session.commit()
+    records = session.query(GithubCloneStats).all()
+    assert previous_record_length - len(records) == 1
+    fetch_and_store_github_clones(organization_name=ORGANIZATION_NAME, repo_name=REPO_NAME, github_token=GH_TOKEN,
+                                  db_parameters=db_params, is_test=True)
+    records = session.query(GithubCloneStats).all()
+    assert len(records) == previous_record_length
+    today_record = session.query(GithubCloneStats).filter_by(clone_date=datetime.today())
+    assert not today_record.first()
```
Comment (halilozanakgul) [inline packaging_automation/tests/test_github_stats_collector.py:44]: Maybe we could also look at the `GithubCloneStatsTransactionsMain` table too.

**Review by halilozanakgul — CHANGES_REQUESTED** [review]
I think we haven't been tracking `uniques` but I think it is good that we add it.

**Review by halilozanakgul — APPROVED** [review]

File: packaging_automation/bash/daily-statistics-job.sh [hunk 4]
```diff
@@ -4,9 +4,14 @@
...
-        --db_user_name "${DB_USER_NAME}" --db_password "${DB_PASSWORD}" --db_host_and_port "${DB_HOST_AND_PORT}" \
-        --db_name "${DB_NAME}"
 
+if [[ ${JOB_NAME} == 'docker_pull_citus' ]]; then
+  python -m packaging_automation.docker_statistics_collector --repo_name citus \
+    --db_user_name "${DB_USER_NAME}" --db_password "${DB_PASSWORD}" --db_host_and_port "${DB_HOST_AND_PORT}" \
+    --db_name "${DB_NAME}"
+elif [[ ${JOB_NAME} == 'github_clone_citus' ]]; then
+  [ -z "${GH_TOKEN:-}" ] && echo "GH_TOKEN should be non-empty value" && exit 1
+  python -m packaging_automation.github_stats_collector --repo_name citus \
```
[the remark this answers is not in the document]
Comment (gurkanindibay) [reply packaging_automation/bash/daily-statistics-job.sh:14]: Fixed

File: packaging_automation/github_statistics_collector.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    fetch_time = datetime.now()
+
+    main_transaction = GithubCloneStatsTransactionsMain(fetch_time=fetch_time, count=contents['count'],
+                                                        repo_name=repo_name, uniques=contents['uniques'])
+    for daily_record in contents['clones']:
+        detail_transaction = GithubCloneStatsTransactionsDetail(clone_date=daily_record.timestamp,
+                                                                count=daily_record.count, uniques=daily_record.uniques)
+        main_transaction.details.append(detail_transaction)
+        # current date's record is skipped since statistics continue to change until end of the day
+        # stat record will not be added if it exists in
```
[the remark this answers is not in the document]
Comment (gurkanindibay) [reply packaging_automation/github_statistics_collector.py:77]: Fixed

**Pull request merged**